docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -357,23 +367,107 @@ spec:
+  [machine-health-checking proposal](https://github.com/kubernetes-sigs/cluster-api/blob/11485f4f817766c444840d8ea7e4e7d1a6b94cc9/docs/proposals/20191030-machine-health-checking.md)
+  for additional details. When there are multiple machines that are marked for remediation, the oldest one will be remediate first.
+  
+- Following rules should be satisfied in order to start remediation
+  - The cluster MUST have at least 3 machines, because this is the smallest cluster size that allows any etcd failure tolerance.
+  - The number of replicas MUST be equal to or greater than the desired replicas. This rule ensures that when the cluster
+    is missing replicas, we skip remediation and instead perform regular scale up/rollout operations first. 
+  - The cluster MUST have no deleting machines. This rule prevents KCP taking actions while the cluster is in a transitional state.
+  - Remediation MUST preserve etcd quorum. This rule ensures that we will not remove a member that would result in etcd
+    losing a majority of members and thus become unable to field new requests.
+
+- When all the conditions for stating remediaiton are satisfied, KCP temporarily suspend any operation in progress
+  in order to perform remediation. 
+- Remediation will be performed by issuing a delete on the unhealthy machine; after deleting the machine, KCP
+  will restore the target number of machines by triggering a scale up (current replicas<desired replicas) and then
+  eventually restart the rollout action.
+  See [why delete and recreate](#why-delete-and-recreate) for an explanation about why KCP should remove the member first and then add its replacement.
+
+###### Why delete and recreate
+
+When replacing a KCP machine the most critical component to be taken into account is etcd, and
+according to the [etcd documentation](https://github.com/etcd-io/etcd/blob/master/Documentation/faq.md#should-i-add-a-member-before-removing-an-unhealthy-member),
+it's important to remove an unhealthy etcd member first and then add its replacement:
+
+_etcd employs distributed consensus based on a quorum model; (n/2)+1 members, a majority, must agree on a proposal before
+it can be committed to the cluster. These proposals include key-value updates and membership changes. This model totally
+avoids any possibility of split brain inconsistency. The downside is permanent quorum loss is catastrophic._
+
+_How this applies to membership: If a 3-member cluster has 1 downed member, it can still make forward progress because
+the quorum is 2 and 2 members are still live. However, adding a new member to a 3-member cluster will increase the quorum
+to 3 because 3 votes are required for a majority of 4 members. Since the quorum increased, this extra member buys nothing
+in terms of fault tolerance; the cluster is still one node failure away from being unrecoverable._
+
+_Additionally, that new member is risky because it may turn out to be misconfigured or incapable of joining the cluster.
```

Comment:
If the 4th member could not join the cluster, that means it did not become a member? So the quorum is still 2.

```suggestion
_Additionally, that new member is risky because it may turn out to be misconfigured.
```

If the 4th one did not join to the cluster, then it is not a member and the quorum is still 2?

Reply:
The 4th etcd node should be Added to the member list before the new member actually starts.
Given that his operation alters the quorom before the join, we should consider as well the case join fails.
(BTW, this is an extract from the etcd docs)